packages/sdk-node/src/__tests__/middleware.test.ts

```diff
@@ -7,6 +7,8 @@ import type { WithAuthProp } from '../types';
 
 const mockNext = jest.fn();
 
+const createRequest = () => ({ url: 'http://localhost:3000', cookies: {}, headers: {} } as Request);
```

Comment:
@nikosdouvlis introducing the `buildRequestUrl` usage in `sdk-node` package causes an `invalid URL` error to be thrown in the tests since we are using the `req.url` passed in the tests to generate a URL which does not have `url` populated.
To resolve this issue I have created the `createRequest` function to construct a Request mock object with a `url` property.

Reply:
However, is this reasonable? I am using Clerk and NestJS, and I found an error after upgrading Clerk. By default, req.url of Express is not populated.

![image](https://github.com/clerkinc/javascript/assets/5787117/b596b07b-a9e9-4532-be6e-db8e821b3cf2)

To make it work. I must do hacking like this.   


![image](https://github.com/clerkinc/javascript/assets/5787117/68ad4874-d626-44d5-b940-0d8d4118cad6)

Reply:
@larryro thank you for your feedback.
I will add a test with a relative `url` as `request.url` and fix this. The issue is caused by https://github.com/clerkinc/javascript/blob/4ff4b716fdb12b18182e506737afafc7dbc05604/packages/sdk-node/src/authenticateRequest.ts#L45